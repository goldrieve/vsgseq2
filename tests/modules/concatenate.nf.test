nextflow_process {

    name "Test Process CONCATENATE_VSGS"
    script "modules/concatenate.nf"
    process "CONCATENATE_VSGS"

    test("Should run without failures") {

        when {
            params {
                vsg_db = "$projectDir/data/blastdb/concatAnTattb427.fa"
            }
            process {
                """
                input [0] = "$projectDir/tests/data/latesingle_ORF_VSGs.fasta"
                input [1] = params.vsg_db
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
