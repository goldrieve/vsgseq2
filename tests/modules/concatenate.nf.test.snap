{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "concatenated_vsgs.fasta:md5,0282cea538b8883720e9e77a4a8625f0"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-21T14:48:54.6178238"
    }
}